uf3/regression/optimize.py

```diff
@@ -0,0 +1,241 @@
+        knot_spacing_2b (float): knot_spacing for 2-body interaction to create the HDF5 file
+        knot_spacing_3b (float): knot_spacing for 3-body interaction to create the HDF5 file
+        leading_trim (float): number of basis functions at leading edge 
+            to suppress. Useful for ensuring smooth cutoffs.
+        trailing_trim (float): number of basis functions at trailing edge 
+            to suppress. Useful for ensuring smooth cutoffs.
+
+    Returns:
+        bspline_config: bspline.BSplineBasis 
+    """
+    #if rmin_2b!=0:
+    #    raise ValueError("Currrent version is only tested for rmin=0")
+    
+    #if rmin_3b!=0:
+    #    raise ValueError("Currrent version is only tested for rmin=0")
+
+    if not (np.isclose((rmax_2b-rmin_2b)%knot_spacing_2b,knot_spacing_2b) or \
+            np.isclose((rmax_2b-rmin_2b)%knot_spacing_2b,0)):
+
+            raise ValueError("Provided rmax_2b does not conatin integer number of \n\
+                knots, seperated by knot_spacing_2b")
+
+    if not (np.isclose((rmax_3b-rmin_3b)%knot_spacing_3b,knot_spacing_3b) or \
+            np.isclose((rmax_3b-rmin_3b)%knot_spacing_3b,0)):
+
+            raise ValueError("Provided rmax_3b does not conatin integer number of \n\
+                knots, seperated by knot_spacing_3b")
+
+    if leading_trim!=0:
+        raise ValueError("Currrent version is only tested for leading_trim=0")
+
+    if trailing_trim!=3:
+        raise ValueError("Currrent version is only tested for trailing_trim=3")
+
+    rmax_3b_double = rmax_3b*2
+    if not(np.isclose((rmax_3b_double%knot_spacing_3b),0) or \
+            np.isclose((rmax_3b_double%knot_spacing_3b),knot_spacing_3b)):
+        raise ValueError("Provided rmax_3b contains integer number of knots sperated \n\
+                by knot_spacing_3b, but rmax_3b_double does not. Consider changing rmax_3b \n\
+                to "+str(rmax_3b+knot_spacing_3b))
+
+    reso_2b = round((rmax_2b - rmin_2b)/knot_spacing_2b)
+    reso_3b = round((rmax_3b - rmin_3b)/knot_spacing_3b)
+
+    reso_3b_double = round((rmax_3b_double - rmin_3b)/knot_spacing_3b)
+    
+    r_min_map = {i:rmin_2b for i in chemical_system.interactions_map[2]}
+    r_min_map.update({i:[rmin_3b,rmin_3b,rmin_3b] for i in chemical_system.interactions_map[3]})
+    
+    r_max_map = {i:rmax_2b for i in chemical_system.interactions_map[2]}
+    r_max_map.update({i:[rmax_3b,rmax_3b,rmax_3b_double] for i in chemical_system.interactions_map[3]})
+    
+    resolution_map = {i:reso_2b for i in chemical_system.interactions_map[2]}
+    resolution_map.update({i:[reso_3b,reso_3b,reso_3b_double] for i in chemical_system.interactions_map[3]})
+
+    bspline_config = bspline.BSplineBasis(chemical_system, r_min_map=r_min_map,
+            r_max_map=r_max_map,resolution_map=resolution_map,
+            trailing_trim=trailing_trim,leading_trim=leading_trim)
+
+    return bspline_config
+
+
+def get_possible_lower_cutoffs(original_bspline_config):
```

Comment:
I do not like this function name. I suggest removing `possible` as it makes the function's use confusing.